cli/src/add_url.rs

```diff
@@ -0,0 +1,31 @@
+// File: add_url.rs
+use color_eyre::Result;
+use db::PgPool;
+use uuid::Uuid;
+
+use crate::check_auth_status;
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug)]
+pub enum AddUrlOutcome {
+    Created(String),
+    Existing(String),
+}
+
+pub async fn add_url(
+    pool: &PgPool,
+    url: &str,
+    user_id: Uuid,
+    allow_existing: bool,
+) -> Result<AddUrlOutcome> {
+    check_auth_status(pool).await?;
+
+    let result = db::add_url(pool, url, user_id, &allow_existing).await;
+    match result {
+        Ok(page) if page.url.contains("re-adding is allowed") => {
+            Ok(AddUrlOutcome::Existing(page.url.to_string()))
+        }
+        Ok(page) => Ok(AddUrlOutcome::Created(page.url.to_string())),
```

Comment:
Ahh! Nice adding this `AddUrlOutcome` enum in!

Would it be possible to move this enum on lever lower and have it be returned by `db::add_url`?
Cause this bit here where we have to check the string to see if its existing or created is the bit I was trying to avoid really lol!

Since the code in `db::add_url` already 'knows' if its a create or existing it should be able to return the specific answer there, and then these other places can use that return type instead of parsing the strings like this!

Does that make sense?

Reply:
Makes sense